lib/cisco_node_utils/ntp_server.rb

```diff
@@ -66,20 +64,41 @@ def ==(other)
 
     def create
       config_set('ntp_server', 'server', state: '', ip: @ntpserver_id,
-                  prefer: @ntpserver_prefer ? 'prefer' : '')
+                  prefer: @prefer == 'true' ? 'prefer' : '',
+                  vrf: @vrf ? "use-vrf #{@vrf}" : '',
+                  key: @key ? "key #{@key}" : '',
+                  minpoll: @minpoll ? "minpoll #{@minpoll}" : '',
+                  maxpoll: @maxpoll ? "maxpoll #{@maxpoll}" : '')
     end
 
     def destroy
       config_set('ntp_server', 'server',
-                 state: 'no', ip: @ntpserver_id, prefer: '')
+                 state: 'no', ip: @ntpserver_id, prefer: '', vrf: '',
+                 key: '', minpoll: '', maxpoll: '')
     end
 
     def name
       @ntpserver_id
     end
 
+    def key
```

Comment:
The following is simpler and will get rid of the rubocop errors.

`attr_reader :key, :maxpoll, :minpoll, :prefer, :vrf`